src/libostree/ostree-repo-commit.c

```diff
@@ -406,6 +410,7 @@ add_size_index_to_metadata (OstreeRepo        *self,
           e_size = g_hash_table_lookup (self->object_sizes, e_checksum);
           _ostree_write_varuint64 (buffer, e_size->archived);
           _ostree_write_varuint64 (buffer, e_size->unpacked);
+          g_string_append_c(buffer, (guchar) e_size->objtype);
```

Comment:
Come to think of it, do we really want to write the objtype directly into the sizes variant? It should be fine if the OstreeObjectType values are globally unique, but if e.g. values 5 and 6 are reused for something else upstream at some point we could be in trouble